packages/fxa-content-server/app/scripts/views/post_verify/finish_account_setup/set_password.js

```diff
@@ -73,7 +73,14 @@ class SetPassword extends FormView {
       const token = this._verificationInfo.get('token');
       return account.isPasswordResetComplete(token).then((isComplete) => {
         if (isComplete) {
-          return this.navigateAway(this._verificationInfo.get('redirectUrl'));
+          return account.fetchSubscriptionPlans().then((plans) => {
+            const productId = this._verificationInfo.get('productId');
+            const plan = plans.find((p) => p.product_id === productId);
+            const url = plan
+              ? plan.product_metadata.downloadURL
+              : 'https://mozilla.org';
```

Comment:
@vbudhram @chenba I think it's better to _pull_ the redirect url than to push it. We don't want an open redirect so we need to validate `redirectUrl`, but against what... a list of valid urls... which we get from the server... 

I think you see where I'm going.

Reply:
@dannycoates Looks like you updated to fetch the url, which I think works better 👍🏽